### PR TITLE
Prepare for Gradle 5.0

### PR DESCRIPTION
**Pull request by oehme**

### CODE REVIEW

File: subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildSymlinkHandlingIntegrationTest.groovy [hunk 1]
```diff
@@ -131,8 +131,8 @@ task work {
         expect:
         executer.expectDeprecationWarning().withFullDeprecationStackTraceDisabled()
         succeeds("work")
-        output.contains """Registering invalid inputs and outputs via TaskInputs and TaskOutputs methods has been deprecated. This is scheduled to be removed in Gradle 5.0. A problem was found with the configuration of task ':work'.
- - File '$link' specified for property '\$1' does not exist."""
+        output.contains """Registering invalid inputs and outputs via TaskInputs and TaskOutputs methods has been deprecated."""
+        outputContains """- File '$link' specified for property '\$1' does not exist."""
```
Comment (marcphilipp) [inline subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalBuildSymlinkHandlingIntegrationTest.groovy]: Nit: 2x `outputContains` with a single line string (`"..."`).

File: subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputPropertiesIntegrationTest.groovy [hunk 2]
```diff
@@ -619,7 +624,8 @@ task someTask(type: SomeTask) {
         expect:
         executer.expectDeprecationWarning().withFullDeprecationStackTraceDisabled()
         succeeds "test"
-        output.contains """Registering invalid inputs and outputs via TaskInputs and TaskOutputs methods has been deprecated. This is scheduled to be removed in Gradle 5.0. A problem was found with the configuration of task ':test'.
+        output.contains """Registering invalid inputs and outputs via TaskInputs and TaskOutputs methods has been deprecated."""
+        outputContains """A problem was found with the configuration of task ':test'.
```
Comment (marcphilipp) [inline subprojects/core/src/integTest/groovy/org/gradle/api/tasks/TaskInputPropertiesIntegrationTest.groovy]: Ditto for this file.

File: subprojects/ide/src/crossVersionTest/groovy/org/gradle/plugins/ide/tooling/r50/ToolingApiIdeaModelCrossVersionSpec.groovy [hunk 3]
```diff
@@ -22,13 +22,17 @@ import org.gradle.integtests.tooling.fixture.ToolingApiVersion
...
+import org.junit.Ignore
 
 /**
  * NOTE: Starting with Gradle 5.0 the contract of IdeaModule#sourceDirs and IdeaModule#testSourceDirs changes in
  * a way that the resource directories are excluded.
  */
 @ToolingApiVersion(">=5.0")
 @TargetGradleVersion(">=5.0")
+@Ignore
+@ToBeImplemented
```
Comment (marcphilipp) [inline subprojects/ide/src/crossVersionTest/groovy/org/gradle/plugins/ide/tooling/r50/ToolingApiIdeaModelCrossVersionSpec.groovy]: Do we have an issue for that?

File: subprojects/plugins/src/test/groovy/org/gradle/api/internal/tasks/DefaultSourceSetContainerTest.groovy [hunk 4]
```diff
@@ -62,13 +62,13 @@ class DefaultSourceSetContainerTest extends Specification {
         name << invalidNames
     }
 
-    void assertForbidden(name, message) {
+    void assertForbidden(String name, String message) {
         if (name == '') {
-            assert message == "The name is empty. This has been deprecated and is scheduled to be removed in Gradle 5.0."
+            assert message.concat("The name is empty. This has been deprecated")
```
Comment (marcphilipp) [inline subprojects/plugins/src/test/groovy/org/gradle/api/internal/tasks/DefaultSourceSetContainerTest.groovy]: `contains`, not `concat`, right?

File: subprojects/plugins/src/test/groovy/org/gradle/api/internal/tasks/DefaultSourceSetContainerTest.groovy [hunk 5]
```diff
@@ -62,13 +62,13 @@ class DefaultSourceSetContainerTest extends Specification {
...
         } else if (name.contains("" + forbiddenLeadingAndTrailingCharacter)) {
-            assert message == """The name '${name}' starts or ends with a '.'. This has been deprecated and is scheduled to be removed in Gradle 5.0."""
+            assert message.contains("""The name '${name}' starts or ends with a '.'. This has been deprecated""")
         } else {
-            assert message == """The name '${name}' contains at least one of the following characters: [ , /, \\, :, <, >, ", ?, *, |]. This has been deprecated and is scheduled to be removed in Gradle 5.0."""
+            assert message.contains("""The name '${name}' contains at least one of the following characters: [ , /, \\, :, <, >, ", ?, *, |]. This has been deprecated""")
```
Comment (marcphilipp) [inline subprojects/plugins/src/test/groovy/org/gradle/api/internal/tasks/DefaultSourceSetContainerTest.groovy]: Nit: No need for `"""`.

File: subprojects/ide/src/crossVersionTest/groovy/org/gradle/plugins/ide/tooling/r50/ToolingApiIdeaModelCrossVersionSpec.groovy (hunk 3 above)
Comment (oehme) [reply subprojects/ide/src/crossVersionTest/groovy/org/gradle/plugins/ide/tooling/r50/ToolingApiIdeaModelCrossVersionSpec.groovy]: https://github.com/gradle/gradle/issues/6312

File: subprojects/plugins/src/test/groovy/org/gradle/api/internal/tasks/DefaultSourceSetContainerTest.groovy (hunk 4 above)
Comment (oehme) [reply subprojects/plugins/src/test/groovy/org/gradle/api/internal/tasks/DefaultSourceSetContainerTest.groovy]: yes

File: subprojects/ide/src/crossVersionTest/groovy/org/gradle/plugins/ide/tooling/r50/ToolingApiIdeaModelCrossVersionSpec.groovy (hunk 3 above)
Comment (marcphilipp) [reply subprojects/ide/src/crossVersionTest/groovy/org/gradle/plugins/ide/tooling/r50/ToolingApiIdeaModelCrossVersionSpec.groovy]: Should we reference it here, e.g. using `@Issue`?

**Pull request merged**